frontend/src/pages/RunList.tsx

```diff
@@ -176,7 +179,101 @@ class RunList extends React.Component<RunListProps, RunListState> {
     }
   }
 
-  private async _loadRuns(request: ListRequest): Promise<string> {
+  public _nameCustomRenderer(value: string, id: string): JSX.Element {
+    return <Link className={commonCss.link} onClick={(e) => e.stopPropagation()}
+      to={RoutePage.RUN_DETAILS.replace(':' + RouteParams.runId, id)}>{value}</Link>;
+  }
+
+  public _pipelineCustomRenderer(pipelineInfo: PipelineInfo, id: string): JSX.Element {
+    // If the getPipeline call failed or a run has no pipeline, we display a placeholder.
+    if (!pipelineInfo || (!pipelineInfo.showLink && !pipelineInfo.id)) {
+      return <div>-</div>;
+    }
+    const search = new URLParser(this.props).build({ [QUERY_PARAMS.fromRunId]: id });
+    const url = pipelineInfo.showLink ?
+      RoutePage.PIPELINE_DETAILS.replace(':' + RouteParams.pipelineId + '?', '') + search :
+      RoutePage.PIPELINE_DETAILS.replace(':' + RouteParams.pipelineId, pipelineInfo.id || '');
+    return (
+      <Link className={commonCss.link} onClick={(e) => e.stopPropagation()}
+        to={url}>
+        {pipelineInfo.showLink ? 'View pipeline' : pipelineInfo.displayName}
```

Comment:
nit: I think maybe `[View pipeline]` would be better? something to help distinguish it visually from a normal pipeline name?

Or maybe we don't actually care about that

Reply:
I think we should distinguish them too. I'll add @ajayalfred here for follow up, but for now I think we can merge this whenever it's ready until we hear back from him.

Reply:
I'll check with Jordan and comment back here.